### DISCUSSION TITLE
Firmware retract.. No setting screen? .. Wrong speeds?

### DISCUSSION THREAD

**Discussion by mikeoverbay** [Jyers/Marlin#1885] [General Discussion]
There seems to be no screen for setting the firmware retract once it's enabled.
I have been using my Ender 3 Pro for a year with hardware retract. It's easier than trial and error test prints and all the wasted time reposting slices, uploading and cleaning the bed.

How hard is it to add this screen? I looked at the source and there is nothing I can find.

Speeds is ok. My bad.

**Comment by Cal-LF** [comment]
What is firmware retract?

**Reply by mikeoverbay** [reply]
It is a feature you can enable in the firmware.
What it does is handle the retraction based on the settings you set as defaults for distance, speed, unretract speed and extra un-retraction amount.
On my Ender 3 Pro.. I can adjust these while its printing on the screen to reduce stringing. There is no such screen on the V2.
I can adjust the setting using [M207 and M208](https://marlinfw.org/docs/gcode/M207.html) in Octoprint's console
I think it makes the gcode more portable between different machines. I don't have to re-slice because one machine has a direct drive extruder.

**Comment by vernonbc** [comment]
I just updated my E3V2 firmware to the latest 2.1.0. This is my first "compile" and I must say it was quite easy as long as you follow the tutorials..
I noticed that the firmware now has "firmware retraction". Will these settings overide my slicer settings or will these values be added on top of my slicer settings making a 4mm retract a 10mm retract?  I see no way of activating or deactivating this feature. At the moment I prefer to use my slicer settings until I am more familiar with the new firmware
Thankyou for the help.

**Reply by Cal-LF** [reply]
It is activated in the slicer under Printer Settings. Firmware must also be activated and compiled.
Reference - https://www.youtube.com/watch?v=T4N59u4wm-Y
You can get the "new features development" to compile here. https://github.com/tititopher68-dev2/Marlin/tree/new_feature
which includes hardware retraction menu.

**Reply by Cal-LF** [reply]
Is this the same as hardware retraction?
